### PR TITLE
Prepare beta 1.49.0

### PR DESCRIPTION
**Pull request by emilyalbini**
This PR sets everything up for beta 1.49.0, and backports the following PRs to it:

* #79107 - build-manifest: strip newline from rustc version 
* #78986 - Avoid installing external LLVM dylibs 
* #79074 - Install CI llvm into the library directory 
* #78364 - Update RELEASES.md for 1.48.0
* #77939 - Ensure that the source code display is working with DOS backline 
 
r? @ghost
cc @rust-lang/release

### CODE REVIEW

File: src/stage0.txt [hunk 1]
```diff
@@ -12,15 +12,15 @@
 # source tarball for a stable release you'll likely see `1.x.0` for rustc and
 # `0.(x+1).0` for Cargo where they were released on `date`.
 
-date: 2020-10-16
-rustc: beta
-cargo: beta
+date: 2020-11-16
+rustc: 1.48.0
+cargo: 1.48.0
```
Comment (Keruspe) [inline src/stage0.txt:17]: Shouldn't this be 0.49.0?

Comment (pietroalbini) [reply src/stage0.txt:17]: https://github.com/rust-lang/rust/pull/77336 changed the dist code to always use the Rust version number for stable tarballs, even for tools like Cargo. Opened https://github.com/rust-lang/rust/pull/79133 to remove the `cargo` field from `src/stage0.txt` :)

**Comment by pietroalbini** [comment]
Hmm, what. Running `./x.py test --stage 2 src/test/ui-fulldeps/` on my machine passes all the tests :/

**Comment by Mark-Simulacrum** [comment]
@bors r+

Cherry-picked a bump of the stderr file for that test. I suspect this is the same problem I saw last time, but I haven't had a chance to look into it much. My loose recollection is there's some channel dependency, but I don't recall why I concluded as such.

**Comment by Mark-Simulacrum** [comment]
Ok, I think I have an understanding of why this is an error. https://github.com/rust-lang/rust/pull/75914 merged in the 1.49 cycle (the beta this PR kicks off). RLS on aarch64 windows never compiled, because winapi 0.2 isn't compatible with aarch64 (it [defines some types only on x86 and x86_64](https://github.com/retep998/winapi-rs/blob/77588710a683e1a4ec085b0f3d6c8939ecc7f030/src/basetsd.rs#L54-L57)). We missed this earlier because toolstate is only looked at by default for x86_64 on the dashboard at https://rust-lang-nursery.github.io/rust-toolstate/.

I'm not sure what the right step here is. I suspect the best bet is probably to just disable RLS on aarch64 Windows, though we may encounter the same problem with other tools (haven't checked). I will try to implement that later today or someone else can take a stab.

**Comment by cuviper** [comment]
RLS has an issue for aarch64 windows: https://github.com/rust-lang/rls/issues/1693

**Comment by Mark-Simulacrum** [comment]
@bors r+

Pushed a commit that should just permit the failure of RLS on aarch64 Windows, which should fix this. I think for now that's fine.

**Comment by mati865** [comment]
I haven't noticed it earlier but you might want to include https://github.com/rust-lang/rust/pull/79141 if this PR fails again.

**Pull request merged**